File: codility/lessons/2 - arrays/odd occurrences in array/odd_occurences_in_array.java
// Source: https://app.codility.com/programmers/lessons/2-arrays/odd_occurrences_in_array/

import java.util.Arrays;

class Solution {

    public int solution(int[] A) {
        Arrays.sort(A);
        int unpairedElement = A[0];
        boolean isElementPaired = false;

        for (int i = 1, N = A.length; i < N; i++)
            if (A[i] == unpairedElement)
                isElementPaired = !isElementPaired;
            else if (isElementPaired) {
                isElementPaired = false;
                unpairedElement = A[i];
            }
            else
                break;

        return unpairedElement;
    }
}

File: codility/lessons/2 - arrays/odd occurrences in array/odd_occurences_in_array.go
// Source: https://app.codility.com/programmers/lessons/2-arrays/odd_occurrences_in_array/

package solution

import "sort"

func Solution(A []int) int {
    sort.Ints(A)
    unpairedElement := A[0]
    isElementPaired := false

    for i, N := 1, len(A); i < N; i++ {
        if A[i] == unpairedElement {
            isElementPaired = !isElementPaired
        } else if isElementPaired {
            isElementPaired = false
            unpairedElement = A[i]
        } else {
            break
        }
    }

    return unpairedElement
}
